app/_components/WebappFrame.tsx

```diff
@@ -69,9 +69,19 @@ const WebappFrame = ({ dotrainText, deploymentOption }: props) => {
 
 	const [error, setError] = useState<string | React.ReactElement | null>(null);
 	const [inputText, setInputText] = useState<string>('');
+	const [isInternalUpdate, setIsInternalUpdate] = useState(false);
 
 	const searchParams = useSearchParams();
 
+	const updateUrl = async (updatedState: FrameState) => {
+		setIsInternalUpdate(true);
+		const url = new URL(window.location.href);
+		const jsonString = JSON.stringify(updatedState);
+		const compressed = await compress(jsonString);
+		url.searchParams.set('currentState', compressed);
+		await window.history.replaceState({}, '', url);
```

Comment:
shouldn't this use push state? https://developer.mozilla.org/en-US/docs/Web/API/History/pushState